src/tools/fuzzing/heap-types.cpp

```diff
@@ -47,6 +47,13 @@ struct HeapTypeGeneratorImpl {
   using HeapTypeKind = std::variant<BasicKind, SignatureKind, DataKind>;
   std::vector<HeapTypeKind> typeKinds;
 
+  // For each type, the index one past the end of its recursion group, used to
+  // determine what types could be valid children.
+  std::vector<Index> recGroupEnds;
```

Comment:
One past the end feels a little odd here. Maybe it's because it's not a pair, it's a single value?

Reply:
Another way to look at it is cumulative sizes of the current and all prior rec groups for each index.